kuma/wiki/admin.py

```diff
@@ -211,30 +212,36 @@ def topic_sibling_documents_link(obj):
         reverse('admin:wiki_document_changelist', args=[]),
         'parent_topic__exact=%s' % (obj.parent_topic.id)
     )
-    what = (count == 1) and 'sibling' or 'siblings'
-    return '<a href="%s">%s&nbsp;%s</a>' % (link, count, what)
+    what = 'sibling' if count == 1 else 'siblings'
+    return format_html('<a href="{}">{}&nbsp;{}</a>', link, count, what)
 
 
-topic_sibling_documents_link.allow_tags = True
 topic_sibling_documents_link.short_description = "Sibling Documents"
 
 
 def document_link(obj):
     """Public link to the document"""
     link = obj.get_absolute_url()
-    return ('<a target="_blank" rel="noopener" href="%s">'
-            '<img src="%simg/icons/link_external.png"> View</a>' %
-            (link, settings.STATIC_URL))
+    return format_html(
+        '<a target="_blank" rel="noopener" href="{}"><img src="{}img/icons/link_external.png"> View</a>',
+        link,
+        settings.STATIC_URL,
+    )
 
 
-document_link.allow_tags = True
 document_link.short_description = "Public"
 
 
 def combine_funcs(obj, funcs):
     """Combine several field functions into one block of lines"""
-    out = (x(obj) for x in funcs)
-    return '<ul>%s</ul>' % ''.join('<li>%s</li>' % x for x in out if x)
+    results = []
+
+    for func in funcs:
+        result = func(obj)
+        if result:
+            results.append(format_html('<li>{}</li>', result))
+
+    return format_html('<ul>{}</ul>', mark_safe(''.join(results)))
```

Comment:
It should be possible to significantly condense this function by using `format_html_join`, but that function is not working as expected for me. So I took a more imperative approach here. :(

The following should work, but doesn't.

```python
def combine_funcs(obj, funcs):
    """Combine several field functions into one block of lines"""
    items = (func(obj) for func in funcs)
    list_body = format_html_join('', '<li>{}</li>', (item for item in items if item))
    return format_html('<ul>{}</ul>', list_body)
```

Using that snippet, each list item is rendered as `<li>&lt;</li>` for God knows why. The input (`items`) looks correct. Each item has an appropriate `__html__` attribute which produces the expected string. But running them through `format_html_join` breaks everything. 🤷‍♂.